source/utils.cc

```diff
@@ -41,4 +42,79 @@ void matrix_market_output_file(
   ASSERT(rv != 0, "EpetraExt::RowMatrixToMatrixMarketFile return value is " +
                       std::to_string(rv));
 }
+
+static bool is_integer(const std::string &s)
+{
+  std::istringstream is(s);
+  int i;
+  is >> i;
+  return !is.fail() && is.eof();
+}
+
+static bool is_double(const std::string &s)
```

Comment:
I think `(boost::property_tree::ptree::get_value_optional<T>()` does what you wanted.